script/zonemaster_backend_testagent

```diff
@@ -60,7 +61,7 @@ $opt_outfile //= '/var/log/zonemaster/zonemaster_backend_testagent.out';
 $loglevel    //= 'info';
 $loglevel = lc $loglevel;
 
-$loglevel =~ /^(?:trace|debug|info|inform|notice|warning|warn|error|err|critical|crit|fatal|alert|emergency)$/ or die "Error: Unrecognized --loglevel $loglevel\n";
+die "Error: Unrecognized --loglevel $loglevel\n" unless exists $CanonicalLevelNames{$loglevel};
```

Comment:
Oh, you're right! This means that the usage documentation for `--loglevel` doesn't describe the implementation. Though I'd argue that for pragmatic reasons it's better to update the implementation to match the documentation than vice versa.

The reason I'm saying this is that I think it would be a good thing if we were able to switch to another log adapter implementation without affecting our own public interfaces. The reasonable way to do this it to either define our own level names or to adopt the level names from Log::Any.

As you point out this is not how it works today. A better way would be to employ Log::Any for the level filtering. Unfortunately the design of that feature is flawed. It's not possible to specify a global log level filter. Instead you have to do it in every module you emit log messages from, but this is not acceptable. (There's a [feature request](https://github.com/preaction/Log-Any/issues/15) that would allow setting a global level filter in Log::Any.)

One possibility I haven't fully explored would be to use Log::Any's built-in adapters instead of Log::Dispatch. In principle that wouldn't be an ultimate solution since we'd still rely on the adapters to do the log filtering, but it would have a couple of other desirable consequences. First we could keep referring to the Log::Any log levels in our public interface, the log levels we implement would match the ones we declare, and we could drop a couple of dependencies.

I realize this was supposed to be a small fix, but it kind of hit the tip of an iceberg. We don't need to solve everything here but we should probably create an issue for solving whatever problems remain after this PR.

Reply:
> > Also $CanonicalLevelNames seems to be undocumented. I always think twice before using undocumented features for fear that they could be unstable too.
> 
> I could use a regex from the values in the documentation if you think that is better.

Yes, I think that would be better.